cmd/blockhashes.cpp

```diff
@@ -42,7 +42,10 @@ int main(int argc, char* argv[]) {
 
     try {
         db::EnvConfig db_config{chaindata};
-        auto result_code{stagedsync::stage_blockhashes(db_config)};
+        auto env{db::open_env(db_config)};
+        stagedsync::TransactionManager tm{env};
+        auto data_dir{DataDirectory::from_chaindata(chaindata)};
+        auto result_code{stagedsync::stage_blockhashes(tm, data_dir.get_etl_path())};
```

Comment:
I'd rely more on DataDirectory as :

```
        auto data_dir{DataDirectory::from_chaindata(chaindata)};
        data_dir.create_tree();
        db::EnvConfig db_config{data_dir.get_chaindata_path().string()};
        auto env{db::open_env(db_config)};
        stagedsync::TransactionManager tm{env};
        auto result_code{stagedsync::stage_blockhashes(tm, data_dir.get_etl_path())};
        check_stagedsync_error(result_code);
```

This will make easier to switch back command line argument --chaindata to --datadir

Reply:
> I'd rely more on DataDirectory as :
> 
> ```
>         auto data_dir{DataDirectory::from_chaindata(chaindata)};
>         data_dir.create_tree();
>         db::EnvConfig db_config{data_dir.get_chaindata_path().string()};
>         auto env{db::open_env(db_config)};
>         stagedsync::TransactionManager tm{env};
>         auto result_code{stagedsync::stage_blockhashes(tm, data_dir.get_etl_path())};
>         check_stagedsync_error(result_code);
> ```
> 
> This will make easier to switch back command line argument --chaindata to --datadir

Fair enough; done.